examples/mdl_sdk/dxr/mdl_d3d12/mdl_material_library.h

```diff
@@ -60,13 +60,27 @@ namespace mi { namespace examples { namespace mdl_d3d12
 
         struct Target_entry
         {
+            // Empty constructor for stl containers
+            Target_entry() {};
+
             // Constructor.
             explicit Target_entry(Mdl_material_target* target);
 
+            // Move constructor for stl containers
+            Target_entry(Target_entry&& other) noexcept
+            {
+                // Lock until the target is ready for move operation
+                std::unique_lock<std::mutex> lock(other.m_mutex);
+
+                // Swap the target pointer
+                m_target = other.m_target;
+                other.m_target = nullptr;
+            }
+
             // Destructor.
             ~Target_entry();
 
-            Mdl_material_target* m_target;
+            Mdl_material_target* m_target = nullptr;
             std::mutex m_mutex;
         };
```

Comment:
Basically, everything changed for using value type. see the comment of each line for details.